packages/admin/dashboard/src/components/inputs/country-select/country-select.tsx

```diff
@@ -6,47 +6,45 @@ import {
 } from "react"
 import { useTranslation } from "react-i18next"
 import { countries } from "../../../lib/data/countries"
-import { Select } from "@medusajs/ui"
+import { Combobox } from "../combobox"
 
 export const CountrySelect = forwardRef<
-  HTMLButtonElement,
-  ComponentPropsWithoutRef<typeof Select> & {
+  HTMLInputElement,
+  Omit<ComponentPropsWithoutRef<typeof Combobox>, "options" | "multiple"> & {
     placeholder?: string
     defaultValue?: string
-    onChange?: (value: string) => void
+    allowClear?: boolean
   }
->(({ disabled, placeholder, defaultValue, onChange, ...field }, ref) => {
+>(({ placeholder, defaultValue, allowClear, onChange, value, ...props }, ref) => {
   const { t } = useTranslation()
-  const innerRef = useRef<HTMLButtonElement>(null)
+  const innerRef = useRef<HTMLInputElement>(null)
+
+  useImperativeHandle(ref, () => innerRef.current as HTMLInputElement)
+
+  // Normalize the value: convert to lowercase
+  // Keep empty string as empty string (don't convert to undefined)
+  const normalizedValue = typeof value === "string"
+    ? (value ? value.toLowerCase() : "")
+    : ""
+
+  const handleChange = (newValue: string | undefined) => {
```

Comment:
discussion: wondering if we should reset filed in this case or set it to `null`. If we treat `undefined` as `""` we would need to implement a check for the fields that are "string | null" in each `handleSubmit`.